hummingbot/client/config/xemm_migration.py

```diff
@@ -0,0 +1,91 @@
+        backup_path = conf_dir_path.parent / "conf_backup"
+        if backup_path.exists():
+            errors = [
+                (
+                    f"\nBackup path {backup_path} already exists."
+                    f"\nThe migration script cannot backup you existing"
+                    f"\nconf files without overwriting that directory."
+                    f"\nPlease remove it and run the script again."
+                )
+            ]
+        else:
+            shutil.copytree(conf_dir_path, backup_path)
+            logging.getLogger().info(f"\nCreated a backup of your existing conf directory to {backup_path}")
+    return errors
+
+
+def migrate_xemm_confs():
+    errors = []
+    logging.getLogger().info("\nMigrating strategies...")
+    for child in conf_dir_path_strategy.iterdir():
+        if child.is_file() and child.name.endswith(".yml"):
+            with open(str(child), "r") as f:
+                conf = yaml.safe_load(f)
+            if "strategy" in conf:
+                if conf["strategy"] == "cross_exchange_market_making":
+                    if "active_order_canceling" in conf:
+                        if conf["active_order_canceling"]:
+                            conf["order_refresh_mode"] = {}
+                        else:
+                            conf["order_refresh_mode"] = {
+                                "cancel_order_threshold": conf["cancel_order_threshold"],
+                                "limit_order_min_expiration": conf["limit_order_min_expiration"]
+                            }
+                        conf.pop("active_order_canceling")
+                        conf.pop("cancel_order_threshold")
+                        conf.pop("limit_order_min_expiration")
+
+                    if "use_oracle_conversion_rate" in conf:
+                        if conf["use_oracle_conversion_rate"]:
+                            conf["conversion_rate_mode"] = {}
+                        else:
+                            conf["conversion_rate_mode"] = {
+                                "taker_to_maker_base_conversion_rate": conf["taker_to_maker_base_conversion_rate"],
+                                "taker_to_maker_quote_conversion_rate": conf["taker_to_maker_quote_conversion_rate"]
+                            }
+                        conf.pop("use_oracle_conversion_rate")
+                        conf.pop("taker_to_maker_base_conversion_rate")
+                        conf.pop("taker_to_maker_quote_conversion_rate")
+
+                    if "template_version" in conf:
+                        conf.pop("template_version")
+
+                    try:
+                        config_map = ClientConfigAdapter(CrossExchangeMarketMakingConfigMap(**conf))
+
+                        save_to_yml(child.absolute(), config_map)
+
+                        logging.getLogger().info(f"Migrated conf for {conf['strategy']}")
+                    except Exception as e:
+                        errors.extend((str(e)))
```

Comment:
I think a better alternative would be to add the check `if conf["strategy"] == "cross_exchange_market_making":` on line 76 of `conf_migration.py` and then add the rest of this code in a function in the same file for two reasons: if we iterate over all strategy files on every strategy we want to migrate, this will be inefficient — we can lump it in one iteration; also, with this approach we'll have to add a new migration file for each strategy, but I don't think that anyone would run individual strategy migration files. They'll either use the client UI, or maybe directly run `migrate_configs` or `migrate_strategies_only` from `conf_migration.py`

Reply:
I don't think it's a good idea to put it to the path migration function. If anyone is running the epic branch now, their paths are already migrated, therefore the xemm migration won't occur anymore. Moreover path migration has no connection to the xemm migration, so that's another reason why separate them.

Reply:
✅